pkg/restapi/issuer/operation/operations.go

```diff
@@ -1309,7 +1369,10 @@ func (o *Operation) handleProposeCredential(msg service.DIDCommAction) (issuecre
 	offerCred := prepareOfferCredentialMessage(manifest, fulfillment)
 
 	// save fulfillment for subsequent WACI steps.
-	err = o.saveCredentialFulfillment(msg.Message.ID(), fulfillment)
+	err = o.saveCredentialAttachments(msg.Message.ID(), &credentialAttachments{
+		Fulfilment: fulfillment,
+		Manifest:   manifest,
```

Comment:
why do we need to save manifest for each message ?

Reply:
We wont know which manifest belongs to which transaction when we need them to read credential application, so for now i am persisting it in the store.

Reply:
only save `presentation_definition` if it is present

Reply:
I was just talking to Talwinder about this, and I think the way the manifests should work is that they should be deterministic per each type of flow. That way you have a finite number of Credential Manifests instead of generating a new one for each transaction - right now in Adapter every time a new propose credential message is received, a new Credential Manifest is generated with a unique ID, but otherwise the same data. I would expect that the total number of Credential Manifests used by one issuer should be a fairly small number - small enough to sit in memory.

You can see in the example from the spec https://identity.foundation/credential-manifest/#credential-manifest---all-features-exercised that the manifest has a set, deterministic ID.

The Credential Application and Credential Fulfillment would still have random IDs though, as they're more dynamic types of messages.

Reply:
Since we will have one manifest per scope can set the Credential Manifest ID to be "[scope]+"_id" something like this ?? rather than being uuid.New.String @sudeshrshetty

Reply:
We can only save presentation_definition if it is present. But when we unmarshal to credential manifest, it automatically invokes unmarshal present in credential manifest libraray and validates and failes with "no output descriptor " no  credential manifest ID or Issuer ID " msg.  @DRK3 @sudeshrshetty

Reply:
We definitely need the full Credential Manifest objects - they're needed for validating Credential Applications against them, so the Issuer can make sure the Holder is sending a proper request... but they shouldn't be particularly heavy (and there should be a bounded number of them).

Reply:
@sudeshrshetty @rolsonquadras See here to see how validation against a CM works: https://github.com/hyperledger/aries-framework-go/blob/bb5bedb39f3610a362829bb0b79b8ad2aca71b72/pkg/doc/cm/credentialapplication.go#L89

It uses a lot more than just the `presentation_definition` to validate.

Reply:
I think i found a way to not persist the manifest rather than just read it again based on the profile and txCredScope when its required in the request Credential. Then we wont need to persist the manifest 

Since we have access to the connection Id based on that i can fetch txnCredscope that was used to read manifest